runtime/compiler/compile/J9Compilation.hpp

```diff
@@ -311,6 +311,8 @@ class OMR_EXTENSIBLE Compilation : public OMR::CompilationConnector
 
    TR_RelocationRuntime *reloRuntime() { return _reloRuntime; }
 
+   bool incompleteOptimizerSupportForReadWriteBarriers();
```

Comment:
Is this API really necessary to pollute the compilation object? I see only one use of it and it is very specific.

Reply:
The other optimization using the query are in omr level https://github.com/eclipse/omr/pull/3224.

Reply:
Ah right, I had not considered that. Thanks!